pkg/netpol/connlist/conns_formatter_dot.go

```diff
@@ -34,49 +42,139 @@ func peerNameAndColorByType(peer Peer) (nameLabel, color string, isExternal bool
+	externalPeersLines = append(externalPeersLines, entireClusterLine...)
+	edgeLines = append(edgeLines, exposureEdges...)
+	// 4. sort graph lines
 	sort.Strings(edgeLines)
 	sort.Strings(externalPeersLines)
-	// collect all lines by order
+	// 5. collect all lines by order
 	allLines := []string{dotformatting.DotHeader}
-	allLines = append(allLines, dotformatting.AddNsGroups(nsPeers)...)
+	allLines = append(allLines, dotformatting.AddNsGroups(nsPeers, dotformatting.DefaultNsGroupColor)...)
+	allLines = append(allLines, dotformatting.AddNsGroups(nsRepPeers, representativeObjColor)...)
 	allLines = append(allLines, externalPeersLines...)
 	allLines = append(allLines, edgeLines...)
 	allLines = append(allLines, dotformatting.DotClosing)
 	return strings.Join(allLines, newLineChar), nil
 }
+
+// addConnlistOutputData updates namespace peers groups and returns edge lines and external peers lines from connlist results
+func addConnlistOutputData(conns []Peer2PeerConnection, nsPeers map[string][]string,
+	peersVisited map[string]bool) (eLines, externalPeersLines []string) {
+	edgeLines := make([]string, len(conns))
+	for index := range conns {
+		edgeLines[index] = getEdgeLine(conns[index])
+		externalPeersLines = append(externalPeersLines, addConnlistPeerLine(conns[index].Src(), nsPeers, peersVisited)...)
+		externalPeersLines = append(externalPeersLines, addConnlistPeerLine(conns[index].Dst(), nsPeers, peersVisited)...)
+	}
+	return edgeLines, externalPeersLines
+}
+
+// addConnlistPeerLine if the given peer is not visited yet, adds it to the relevant lines' group (namespace group/ external)
+func addConnlistPeerLine(peer Peer, nsPeers map[string][]string, peersVisited map[string]bool) (externalPeerLine []string) {
+	if !peersVisited[peer.String()] {
+		peersVisited[peer.String()] = true
+		peerLine, isExternalPeer := getPeerLine(peer)
+		if isExternalPeer { // peer that does not belong to a cluster's namespace (i.e. ip/ ingress-controller)
+			externalPeerLine = []string{peerLine}
+		} else { // add to Ns group
+			dotformatting.AddPeerToNsGroup(peer.Namespace(), peerLine, nsPeers)
+		}
+	}
+	return externalPeerLine
+}
+
+// addExposureOutputData gets the exposure-analysis results, updates the namespaces peers groups lines for both real exposed peers and
+// representative peers and returns the exposure edges and entire cluster line (as external peer line)
+func addExposureOutputData(exposureConns []ExposedPeer, peersVisited map[string]bool,
+	nsPeers, nsRepPeers map[string][]string) (entireClusterLine, exposureEdges []string) {
+	representativeVisited := make(map[string]bool, 0) // acts as a set
+	for _, ep := range exposureConns {
+		if !peersVisited[ep.ExposedPeer().String()] { // an exposed peer is a real peer from the manifests,
+			// updated in the real namespaces map
+			exposedPeerLine, _ := getPeerLine(ep.ExposedPeer())
+			dotformatting.AddPeerToNsGroup(ep.ExposedPeer().Namespace(), exposedPeerLine, nsPeers)
+		}
```

Comment:
how is it possible that a real peer is not visited by this stage? (from the original connections list?)

Reply:
possible use-case:  a real peer that is selected by a policy that exposes it to representative peers only on both ingress and egress. (this peer will not appear in the original connlist report)
a test was added (running it twice once without exposure flag and once with , to see that without it does not appear in the report)